plugins/adrv9002.c

```diff
@@ -927,6 +1189,1009 @@ static void load_profile(GtkFileChooser *chooser, gpointer data)
+
+	char filename[64];
+	sprintf(filename, "adrv9002%s", save_type? ".json": ".stream");
+	GtkFileChooser *chooser = GTK_FILE_CHOOSER(gtk_builder_get_object(priv->builder, "save_panel"));
+	gtk_file_chooser_set_current_name(chooser, filename);
+}
+
+static void profile_gen_save_dialog_show(GtkButton* self, gpointer data)
+{
+	struct plugin_private *priv = data;
+	GtkFileChooser *chooser = GTK_FILE_CHOOSER(gtk_builder_get_object(priv->builder, "save_panel"));
+
+	gtk_file_chooser_set_action(chooser, GTK_FILE_CHOOSER_ACTION_SAVE);
+	gtk_file_chooser_set_do_overwrite_confirmation(chooser, TRUE);
+
+	gtk_file_chooser_set_current_folder(chooser, getenv("HOME"));
+	profile_gen_save_type_changed(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, "cb_save_type")), priv);
+
+	gtk_widget_show(GTK_WIDGET(chooser));
+}
+
+static void profile_gen_update_orx(GtkComboBox* self, struct plugin_private *priv)
+{
+	int chann;
+	char widget_str[25];
+	bool tdd_en = atoi(gtk_combo_box_get_active_id(GTK_COMBO_BOX(gtk_builder_get_object(priv->builder, "cb_radio_duplex")))) == 0;
+	bool tx_en;
+
+	for (chann = 0; chann < 2; chann ++) {
+		sprintf(widget_str, "cb_tx_chan%d_en", chann + 1);
+		tx_en = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)));
+
+		sprintf(widget_str, "frame_radio_orx%d", chann + 1);
+		gtk_widget_set_sensitive(GTK_WIDGET(gtk_builder_get_object(priv->builder, widget_str)), tx_en && tdd_en);
+
+		// deactivate if disabled
+		if (!(tx_en && tdd_en)) {
+			sprintf(widget_str, "cb_radio_orx%d_en", chann + 1);
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), false);
+		}
+	}
+}
+
+
+char *extract_value_between(const char *str, const char *begin, const char *end)
+{
+	// if end is empty it will stop at the end of str
+	const char *i1 = strstr(str, begin);
+	if(i1 != NULL)
+	{
+		const size_t pl1 = strlen(begin);
+		const char *i2 = strcmp(end, "") != 0? strstr(i1 + pl1, end): NULL;
+		const size_t mlen = strcmp(end, "") != 0? i2 - (i1 + pl1): (int)(strlen(i1) - pl1);
+		char *ret = malloc(mlen + 1);
+
+		if(ret != NULL)
+		{
+			memcpy(ret, i1 + pl1, mlen);
+			ret[mlen] = '\0';
+			return ret;
+		}
+	}
+
+	return NULL;
+}
+
+static int profile_gen_config_get_from_device(struct adrv9002_config *cfg, gpointer data)
+{
+	struct adrv9002_config default_cfg = lte_lvs_3072_MHz_10();
+	struct plugin_private *priv = data;
+	int ret_value = 0;
+
+	// radio_config
+	radio_config radio_config;
+
+	char buf[1024];
+	char profile_config[512];
+	ret_value = iio_device_attr_read(priv->adrv9002, "profile_config", profile_config, sizeof(profile_config));
+	if (ret_value < 0) {
+		printf("\nFailed to get device attr read %s! error code: %d", "profile_config", ret_value);
+		goto iio_error;
+	}
+
+	// radio_config.ssi_lanes
+	char *ssi_interface = extract_value_between(profile_config, "SSI interface:", "");
+	if (ssi_interface == NULL) {
```

Comment:
nit `if (!ssi_interface)`. Moreover the `else if` is redundant in here. You should have

```
ssi_interface = extract_value_between(profile_config, "SSI interface:", "");
if (!ssi_interface == NULL) {
     printf("Failed to get SSI interface!/n");
     ret_value = -1;
     goto iio_error;
}

/* then you can start with the other diffs */
```

I would also use proper error codes like `-EINVAL` or something....